pinot-common/src/main/java/org/apache/pinot/common/utils/LLCSegmentName.java

```diff
@@ -76,13 +118,7 @@ public static LLCSegmentName of(String segmentName) {
    * Returns whether the given segment name represents an LLC segment.
    */
   public static boolean isLLCSegment(String segmentName) {
-    int numSeparators = 0;
-    int index = 0;
-    while ((index = segmentName.indexOf(SEPARATOR, index)) != -1) {
-      numSeparators++;
-      index += 2; // SEPARATOR.length()
-    }
-    return numSeparators == 3;
+    return of(segmentName) != null;
```

Comment:
Let's only change the last check as `numSeparators == 3 || numSeparators == 4` as it is more efficient